agent/serviceinfobroker/service_info_broker.go

```diff
@@ -174,28 +176,28 @@ func (sib *ServiceInfoBroker) getMongoDBInfo(ctx context.Context, dsn string, fi
 		return &res
 	}
 
-	resp := client.Database("admin").RunCommand(ctx, bson.D{{Key: "getDiagnosticData", Value: 1}})
-	if err = resp.Err(); err != nil {
-		sib.l.Debugf("getMongoDBInfo: failed to runCommand getDiagnosticData: %s", err)
+	mongoVersion, err := agent_version.GetMongoDBVersion(ctx, client)
+	if err != nil {
+		sib.l.Debugf("getMongoDBInfo: failed to get MongoDB version: %s", err)
 		res.Error = err.Error()
 		return &res
 	}
 
-	resp = client.Database("admin").RunCommand(ctx, bson.D{{Key: "buildInfo", Value: 1}})
+	// use hello command for newer MongoDB versions
```

Comment:
I believe that this hello command should be moved to connection checker

Reply:
Seems in this case, we don't need to run `hello` or `getDiagnosticsData` in the service info broker anymore, and we only need to get the version - since that's what we do for other services.

Reply:
right